apis/python/tests/test_dataframe.py

```diff
@@ -461,6 +468,253 @@ def test_bool_enums(tmp_path, data_and_expected_levels):
         assert actual == expect
 
 
+@pytest.mark.parametrize("extend_not_write", [False, True])
+@pytest.mark.parametrize("ordered", [True, False])
+def test_extend_enumeration_values(tmp_path, extend_not_write, ordered):
```

Comment:
another missing test case:  extending the schema with an array that contains nulled positions

Reply:
And another:   send `extend_enumeration` a value array that is a slice of another array.   This will catch mistakes where you fail to account for the Arrow array offset.

Example:

> df.extend_enumeration({ 'colname':  pa.array(['a','b','c'])[1:-1])

Reply:
> another missing test case: extending the schema with an array that contains nulled positions

https://github.com/single-cell-data/TileDB-SOMA/pull/3815/commits/1b24a55eef0e9d5ff7b4696eb7ff8bedc6edfcde